### PR TITLE
use getMaybeAllocationDomain instead of getMaybeRFactorDomain 

### LINKED ISSUE #902: Specifying AllocationDomain causes vectorization error `Vectorized dim for consumer has to be from a contiguous inner most position.`
Currently scheduler doesn't consider `AllocationDomain` during vectorization. So we ended up setting up vectorization on something that's not the inner most dimension.

I run into this repro with point-wise scheduler, but it's likely a wide spread problem.

```
TEST_F(NVFuserTest, VectorizationAllocationDeomainRepro) {
  auto fusion_ptr = std::make_unique<Fusion>();
  auto& fusion = *fusion_ptr;
  FusionGuard fg(fusion_ptr.get());

  const std::vector<int64_t> shape({16, 16, 512, 64});

  auto tv0 = makeContigTensor(4);
  fusion.addInput(tv0);

  auto tv1 = set(tv0);
  fusion.addOutput(tv1);

  std::vector<nvfuser::IterDomain*> aloc_domain;
  aloc_domain.push_back(tv1->axis(0));
  aloc_domain.push_back(tv1->axis(2));
  aloc_domain.push_back(tv1->axis(3));
  aloc_domain.push_back(tv1->axis(1));
  tv1->setAllocationDomain(aloc_domain, true);

  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
  auto t0 = at::randn(shape, options);
  std::vector<c10::IValue> aten_inputs({t0});

  FusionExecutorCache executor_cache(std::move(fusion_ptr));
  auto cg_outputs = executor_cache.runFusionWithInputs(aten_inputs);
}
```

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes #902 

1. for inner most dimension query, we should be using allocation domain instead of rfactor domain.
2. renamed transpose benchmark from `NF_Transpose_XXX` to `NvFuserScheduler_Transpose_XXX`

### CODE REVIEW

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
CI looks green. so I guess this one is good for now~~

File: test/test_allocation_domain.cpp [hunk 1]
```diff
@@ -1053,4 +1053,34 @@ TEST_F(AllocationDomainTest, NHWC2d_To_NHWC2d_cacheFork_CUDA) {
   testValidate(&fusion, cg_outputs, {t0}, {t0, t0}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, AllocationDomainVectorizationIssue902) {
```
Comment (jjsjann123) [inline test/test_allocation_domain.cpp:1056]: changes on codegen switched this fusion from pointwise scheduler to transpose scheduler. Which fixed the vectorization on non-inner most dimension issue.

Unfortunately we are not getting vectorization on both read / write. I was aware of that issue already, this is a note that I need to patch that as well.

File: benchmark/transpose.cpp [hunk 2]
```diff
@@ -247,31 +247,31 @@ static void NvFuserScheduler_Transpose(
       ->Unit(benchmark::kMicrosecond)
 
 NVFUSER_TRANSPOSE_SQUARE_RUN(
-    NF_Transpose_Random_fp32_Inner_2D_01_Axis,
+    NvFuserScheduler_Transpose_Random_fp32_Inner_2D_01_Axis,
```
Comment (naoyam) [inline benchmark/transpose.cpp:250]: 👍 

No other benchmark named like `NF_`?

File: csrc/scheduler/utils.cpp [hunk 3]
```diff
@@ -1202,7 +1202,7 @@ IterDomain* projectIdToRFactor(
 } // namespace
 
 IterDomain* innerMostRootDim(TensorView* tv) {
-  const auto& root_domain = tv->getMaybeRFactorDomain();
+  const auto& root_domain = tv->getMaybeAllocationDomain();
```
Comment (naoyam) [inline csrc/scheduler/utils.cpp:1205]: nit: change the function name here too? `innerMostAllocDim`?

**Comment by naoyam** [comment]
Are all uses of `innerMostRootDim` fine with this change? For example, here we are still looking at rfactor domains.

https://github.com/NVIDIA/Fuser/blob/main/csrc/scheduler/utils.cpp#L1330-L1339

File: benchmark/transpose.cpp (hunk 2 above)
Comment (jjsjann123) [reply benchmark/transpose.cpp:250]: Found a few more when comparing benchmark. I'm still cleaning this up.

**Comment by jjsjann123** [comment]
!build

**Comment by jjsjann123** [comment]
I'm sure there's still quite a few places in ContiguousInnerDimensionsMapper that should also refactor to use allocation domain instead. Let's do that in a separate PR?!

At least I'm not seeing any failures and transpose benchmark code doesn't seems to be affected by this change here neither.

**Comment by zasdfgbnm** [comment]
> I'm sure there's still quite a few places in ContiguousInnerDimensionsMapper that should also refactor to use allocation domain instead. Let's do that in a separate PR?!
> 
> At least I'm not seeing any failures and transpose benchmark code doesn't seems to be affected by this change here neither.

Yeah, please don't change `ContiguousInnerDimensionsMapper` in this PR. Vectorization analysis for allocation domain is not mechanical. You need to do propagation in rfactor domain, and project the final result to allocation domain.

File: csrc/scheduler/transpose.cpp [hunk 4]
```diff
@@ -235,7 +235,7 @@ class DomainMap : public pointwise_utils::DomainMap {
     // transformed. So the mapping here would require a new compute at map to be
     // constructed from the updated fusion. We'll revisit this once our id graph
     // refactor is done.
-    auto mapped_id = getMappedRootDimIn(tv, root_dim);
+    auto mapped_id = getMappedAllocDimIn(tv, root_dim);
```
Comment (zasdfgbnm) [inline csrc/scheduler/transpose.cpp:238]: So originally, you were looking for inner domain of group 2 in the global reference's root domain, but now you are looking for it in the allocation domain. For the case where the allocation domain is not a recorder of rFactor, neither will work. But for now, I think it should be fine.

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/scheduler/transpose.cpp (hunk 4 above)
Comment (jjsjann123) [reply csrc/scheduler/transpose.cpp:238]: Thanks for pointing this out.

I like how fancy our allocation domain support is, yet scheduler are like... thank you but no. :laughing:

**Pull request merged**